atakama/rule_engine.py

```diff
@@ -0,0 +1,195 @@
+"""Atakama keyserver ruleset library"""
+import abc
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Dict, Union, TYPE_CHECKING
+import logging
+
+import yaml
+
+from atakama import Plugin
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+log = logging.getLogger(__name__)
+
+
+class RequestType(Enum):
+    DECRYPT = "decrypt"
+    SEARCH = "search"
+    CREATE_PROFILE = "create_profile"
+    ACTIVATE_LOCATION = "activate_location"
+    CREATE_LOCATION = "create_location"
+    RENAME_FILE = "rename_file"
+    SECURE_EXPORT = "secure_export"
+
+
+@dataclass
+class ProfileInfo:
+    profile_id: bytes
+    """Requesting profile uuid"""
+    profile_words: List[str]
+    """Requesting profile 'words' mnemonic"""
+
+
+@dataclass
+class MetaInfo:
+    meta: str
+    """Typically the mount point sub-path of a file."""
```

Comment:
The computer meta str would include the full path from `/`. We could use a different field to split this out if desired. Hypothetically it might make sense to just include the root epub, but I wouldn't expect a plugin to have the ability to translate that yet.